File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int arr[] = new int[26];
        for(char c: s.toCharArray()){
            arr[c - 'a']++;
        } 

        for(int i = 0; i < t.length(); i++){
            char c = t.charAt(i);
            if(arr[c - 'a'] == 0){
                return false;
            }else{
                arr[c - 'a']--;
            }
        }

        for(int i: arr){
            if(i != 0){
                return false;
            }
        }

        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        # freqDict = {}
        # for letter in s:
        #     freqDict[letter] = freqDict.get(letter, 0) + 1
        # for letter in t:
        #     freqDict[letter] = freqDict.get(letter, 0) - 1
            
        # for val in freqDict.values():
        #     if val != 0:
        #         return False
        # return True

        return Counter(s) == Counter(t)

        
        
        